From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 2 Nov 2018 20:01:11 +0100
Subject: Remove passwords keyboard accessory option

Completely rip out #passwords-keyboard-accessory, since the default of
disabled does not seem to be working
---
 chrome/browser/about_flags.cc                                     | 5 -----
 chrome/browser/android/chrome_feature_list.cc                     | 1 -
 chrome/browser/flag_descriptions.cc                               | 6 ------
 chrome/browser/flag_descriptions.h                                | 3 ---
 chrome/browser/password_manager/password_accessory_controller.cc  | 8 +-------
 .../password_manager/core/common/password_manager_features.cc     | 4 ----
 .../password_manager/core/common/password_manager_features.h      | 1 -
 7 files changed, 1 insertion(+), 27 deletions(-)

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -2854,11 +2854,6 @@ const FeatureEntry kFeatureEntries[] = {
     {"password-search", flag_descriptions::kPasswordSearchMobileName,
      flag_descriptions::kPasswordSearchMobileDescription, kOsAndroid,
      FEATURE_VALUE_TYPE(password_manager::features::kPasswordSearchMobile)},
-    {"passwords-keyboard-accessory",
-     flag_descriptions::kPasswordsKeyboardAccessoryName,
-     flag_descriptions::kPasswordsKeyboardAccessoryDescription, kOsAndroid,
-     FEATURE_VALUE_TYPE(
-         password_manager::features::kPasswordsKeyboardAccessory)},
 #endif  // OS_ANDROID
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
     {"passwords-migrate-linux-to-login-db",
diff --git a/chrome/browser/android/chrome_feature_list.cc b/chrome/browser/android/chrome_feature_list.cc
--- a/chrome/browser/android/chrome_feature_list.cc
+++ b/chrome/browser/android/chrome_feature_list.cc
@@ -170,7 +170,6 @@ const base::Feature* kFeaturesExposedToJava[] = {
     &omnibox::kUIExperimentHideSteadyStateUrlSchemeAndSubdomains,
     &password_manager::features::kPasswordExport,
     &password_manager::features::kPasswordSearchMobile,
-    &password_manager::features::kPasswordsKeyboardAccessory,
     &unified_consent::kUnifiedConsent,
     &subresource_filter::kSafeBrowsingSubresourceFilter,
 };
diff --git a/chrome/browser/flag_descriptions.cc b/chrome/browser/flag_descriptions.cc
--- a/chrome/browser/flag_descriptions.cc
+++ b/chrome/browser/flag_descriptions.cc
@@ -1480,12 +1480,6 @@ const char kPasswordSearchMobileName[] = "Password search";
 const char kPasswordSearchMobileDescription[] =
     "Search functionality in password settings.";
 
-const char kPasswordsKeyboardAccessoryName[] =
-    "Add password-related functions to keyboard accessory";
-const char kPasswordsKeyboardAccessoryDescription[] =
-    "Adds password generation button and toggle for the passwords bottom sheet "
-    "to the keyboard accessory. Replaces password generation popups.";
-
 const char kPasswordsMigrateLinuxToLoginDBName[] =
     "Migrate passwords to \"Login Data\"";
 const char kPasswordsMigrateLinuxToLoginDBDescription[] =
diff --git a/chrome/browser/flag_descriptions.h b/chrome/browser/flag_descriptions.h
--- a/chrome/browser/flag_descriptions.h
+++ b/chrome/browser/flag_descriptions.h
@@ -909,9 +909,6 @@ extern const char kPasswordImportDescription[];
 extern const char kPasswordSearchMobileName[];
 extern const char kPasswordSearchMobileDescription[];
 
-extern const char kPasswordsKeyboardAccessoryName[];
-extern const char kPasswordsKeyboardAccessoryDescription[];
-
 extern const char kPasswordsMigrateLinuxToLoginDBName[];
 extern const char kPasswordsMigrateLinuxToLoginDBDescription[];
 
diff --git a/chrome/browser/password_manager/password_accessory_controller.cc b/chrome/browser/password_manager/password_accessory_controller.cc
--- a/chrome/browser/password_manager/password_accessory_controller.cc
+++ b/chrome/browser/password_manager/password_accessory_controller.cc
@@ -158,13 +158,7 @@ PasswordAccessoryController::~PasswordAccessoryController() = default;
 bool PasswordAccessoryController::AllowedForWebContents(
     content::WebContents* web_contents) {
   DCHECK(web_contents) << "Need valid WebContents to attach controller to!";
-  if (vr::VrTabHelper::IsInVr(web_contents)) {
-    return false;  // TODO(crbug.com/865749): Reenable if works for VR keyboard.
-  }
-  // Either #passwords-keyboards-accessory or #experimental-ui must be enabled.
-  return base::FeatureList::IsEnabled(
-             password_manager::features::kPasswordsKeyboardAccessory) ||
-         base::FeatureList::IsEnabled(features::kExperimentalUi);
+  return false;
 }
 
 // static
diff --git a/components/password_manager/core/common/password_manager_features.cc b/components/password_manager/core/common/password_manager_features.cc
--- a/components/password_manager/core/common/password_manager_features.cc
+++ b/components/password_manager/core/common/password_manager_features.cc
@@ -68,10 +68,6 @@ const base::Feature kPasswordImport = {"PasswordImport",
 const base::Feature kPasswordSearchMobile = {"PasswordSearchMobile",
                                              base::FEATURE_ENABLED_BY_DEFAULT};
 
-// Adds password-related features to the keyboard accessory on mobile devices.
-const base::Feature kPasswordsKeyboardAccessory = {
-    "PasswordsKeyboardAccessory", base::FEATURE_DISABLED_BY_DEFAULT};
-
 // Enables the experiment for the password manager to only fill on account
 // selection, rather than autofilling on page load, with highlighting of fields.
 const base::Feature kFillOnAccountSelect = {"fill-on-account-select",
diff --git a/components/password_manager/core/common/password_manager_features.h b/components/password_manager/core/common/password_manager_features.h
--- a/components/password_manager/core/common/password_manager_features.h
+++ b/components/password_manager/core/common/password_manager_features.h
@@ -32,7 +32,6 @@ extern const base::Feature kNewPasswordFormParsingForSaving;
 extern const base::Feature kPasswordExport;
 extern const base::Feature kPasswordImport;
 extern const base::Feature kPasswordSearchMobile;
-extern const base::Feature kPasswordsKeyboardAccessory;
 extern const base::Feature kProtectSyncCredential;
 extern const base::Feature kProtectSyncCredentialOnReauth;
 
-- 
2.11.0

